src/amplitude_experiment/cohort/cohort_download_api.py

```diff
@@ -0,0 +1,93 @@
+import time
+import logging
+import base64
+import json
+from http.client import HTTPResponse
+from typing import Optional
+
+from .cohort import Cohort
+from ..connection_pool import HTTPConnectionPool
+from ..exception import HTTPErrorResponseException, CohortTooLargeException, CohortNotModifiedException
+
+CDN_COHORT_SYNC_URL = 'https://cohort-v2.lab.amplitude.com'
+
+
+class CohortDownloadApi:
+    def __init__(self):
+        self.cdn_server_url = CDN_COHORT_SYNC_URL
+
+    def get_cohort(self, cohort_id: str, cohort: Cohort) -> Optional[Cohort]:
+        raise NotImplementedError
+
+
+class DirectCohortDownloadApi(CohortDownloadApi):
+    def __init__(self, api_key: str, secret_key: str, max_cohort_size: int, cohort_request_delay_millis: int,
+                 debug: bool):
+        super().__init__()
+        self.api_key = api_key
+        self.secret_key = secret_key
+        self.max_cohort_size = max_cohort_size
+        self.__setup_connection_pool()
+        self.cohort_request_delay_millis = cohort_request_delay_millis
+        self.logger = logging.getLogger("Amplitude")
+        self.logger.addHandler(logging.StreamHandler())
+        if debug:
+            self.logger.setLevel(logging.DEBUG)
+
+    def get_cohort(self, cohort_id: str, cohort: Cohort) -> Optional[Cohort]:
```

Comment:
Sorry I meant that the cohort that is passed in as an argument is optional, not the cohort that is returned.